app/src/main/java/org/schabi/newpipe/util/StreamDialogEntry.java

```diff
@@ -33,6 +37,21 @@
                 item.getServiceId(), item.getUploaderUrl(), item.getUploaderName())
     ),
 
+    download(R.string.download, (fragment, item) -> {
+        final int serviceId = item.getServiceId();
+        final String url = item.getUrl();
+
+        ExtractorHelper.getStreamInfo(serviceId, url, true)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(result -> {
+                    if (PermissionHelper.checkStoragePermissions(fragment.getActivity(),
+                            PermissionHelper.DOWNLOAD_DIALOG_REQUEST_CODE)) {
+                        NavigationHelper.openDownloadDialog(
+                                (AppCompatActivity) fragment.getActivity(), result);
+                    }
+                });
```

Comment:
An error handler is missing here. A snackbar should be shown to the user if fetching video info could not be achieved.

Reply:
OK, I'll add that.